cdap-app-templates/cdap-etl/cdap-etl-realtime-app/src/main/java/co/cask/cdap/app/etl/realtime/ETLWorker.java

```diff
@@ -269,6 +306,16 @@ public void run(DatasetContext context) throws Exception {
     }
   }
 
+  private Put constructErrorPut(byte[] rowKey, InvalidEntry entry) throws IOException {
+    Put errorPut = new Put(rowKey);
+    errorPut.add(Constants.ErrorDataset.COLUMN_ERRCODE, entry.getErrorCode());
+    errorPut.add(Constants.ErrorDataset.COLUMN_ERRMSG, entry.getErrorMsg());
+    StructuredRecord record =  (StructuredRecord) entry.getInvalidRecord();
```

Comment:
technically, there is no guarantee that the InvalidRecord is a StructuredRecord, even though this is true for all our transforms. Lets check that it really is a StructuredRecord, then just toString() it if it's not.